user_assigned_identities.tf

```diff
@@ -1,11 +1,21 @@
+locals {
+  # Ensure Managed Identity has required permissions to read basic user information
+  # when the caller wants to create Entra ID Groups. Having only "Owner" on the group
+  # is not sufficient (even though the Terraform Provider docs says so).
+  app_role_assignments_computed = setunion(
+    var.app_role_assignments,
+    toset(length(var.groups) == 0 ? [] : ["User.ReadBasic.All"])
```

Comment:
It wasn't enough to compare `var.groups` to either of `{}` or `null` - I don't really know why. This is why I instead compare the length of the map to check if any groups are being created.